From 1e121de76c5db22615cde6311102cb519aa86c37 Mon Sep 17 00:00:00 2001
From: Aishwarya R <quic_aisr@quicinc.com>
Date: Tue, 15 Mar 2022 16:52:41 +0530
Subject: [PATCH] ath12k: Increasing VDEV_SETUP_TIMEOUT value to 3s

[ 1188.248862] ath12k_pci 0002:01:00.0: failed to synchronize setup for vdev 0 start: -110
[ 1188.248878] ath12k_pci 0002:01:00.0: failed to start vdev 0 addr 00:03:7f:12:0f:0f on freq 2462: -110
Increased timeout value to fix timeout error in ath12k_mac_vdev_setup_sync

Signed-off-by: Aishwarya R <quic_aisr@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 2 ++
 drivers/net/wireless/ath/ath12k/wmi.h | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 6e5b33e..5b11e13 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -738,6 +738,8 @@ static inline int ath12k_mac_vdev_setup_sync(struct ath12k *ar)
 	if (test_bit(ATH12K_FLAG_CRASH_FLUSH, &ar->ab->dev_flags))
 		return -ESHUTDOWN;
 
+	ath12k_dbg(ar->ab, ATH12K_DBG_MAC, "ath12k vdev setup timeout %d\n",
+		   ATH12K_VDEV_SETUP_TIMEOUT_HZ);
 	if (!wait_for_completion_timeout(&ar->vdev_setup_done,
 					 ATH12K_VDEV_SETUP_TIMEOUT_HZ))
 		return -ETIMEDOUT;
diff --git a/drivers/net/wireless/ath/ath12k/wmi.h b/drivers/net/wireless/ath/ath12k/wmi.h
index f0acc52..2723bdb 100644
--- a/drivers/net/wireless/ath/ath12k/wmi.h
+++ b/drivers/net/wireless/ath/ath12k/wmi.h
@@ -2589,7 +2589,7 @@ struct wmi_ssid {
 	u32 ssid[8];
 } __packed;
 
-#define ATH12K_VDEV_SETUP_TIMEOUT_HZ (1 * HZ)
+#define ATH12K_VDEV_SETUP_TIMEOUT_HZ (3 * HZ)
 
 struct wmi_vdev_start_request_cmd {
 	u32 tlv_header;
-- 
2.17.1

